test/extensions/filters/http/alternate_protocols_cache/filter_integration_test.cc

```diff
@@ -0,0 +1,79 @@
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h"
+#include "envoy/extensions/transport_sockets/tls/v3/cert.pb.h"
+
+#include "extensions/transport_sockets/tls/context_config_impl.h"
+#include "extensions/transport_sockets/tls/ssl_socket.h"
+
+#include "test/integration/http_integration.h"
+#include "test/integration/http_protocol_integration.h"
+#include "test/integration/ssl_utility.h"
+
+namespace Envoy {
+namespace {
+
+class FilterIntegrationTest : public HttpProtocolIntegrationTest {
+protected:
+  void initialize() override {
+    const std::string filter = R"EOF(
+name: alternate_protocols_cache
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.alternate_protocols_cache.v3.FilterConfig
+  alternate_protocols_cache_options:
+    name: default_alternate_protocols_cache
+)EOF";
+    config_helper_.addFilter(filter);
+
+    upstream_tls_ = true;
+    config_helper_.configureUpstreamTls(/*use_alpn=*/true, /*http3=*/true);
+
+    HttpProtocolIntegrationTest::initialize();
+  }
+
+  void createUpstreams() override {
+    ASSERT_EQ(upstreamProtocol(), Http::CodecType::HTTP3);
```

Comment:
worth commenting manual creation (that the integration test doesn't handle mixed protocol upstreams very well)

Reply:
Done.